integration_tests/tests/integration/simulation/proxy.rs

```diff
@@ -0,0 +1,168 @@
+use std::{
+    borrow::Cow,
+    sync::{Arc, Mutex, OnceLock},
+};
+
+use regex::bytes::Regex;
+use tracing::error;
+use trillium::{Conn, Handler, Status};
+use trillium_macros::Handler;
+
+/// A [`Handler`] wrapper that can be configured to drop requests or responses.
+#[derive(Handler)]
+pub(super) struct FaultInjectorHandler<H> {
+    #[handler(except = [run, before_send, name])]
+    inner: H,
+    error_before: Arc<Mutex<bool>>,
+    error_after: Arc<Mutex<bool>>,
```

Comment:
Comments may be justified here, i.e. before and after _what_ happens? AIUI `before` induces error before business-logic-containing handler happens, and `after` induces it, well, after?